includes/service-providers/active_campaign/class-newspack-newsletters-active-campaign.php

```diff
@@ -227,23 +281,32 @@ public function test( $post_id, $emails ) {
 				__( 'ActiveCampaign API credentials are missing.', 'newspack-newsletters' )
 			);
 		}
-		$sync_result = $this->sync( get_post( $post_id ) );
-		if ( is_wp_error( $sync_result ) ) {
-			return $sync_result;
+		$post = get_post( $post_id );
+		/** Create disposable campaign for sending a test. */
+		$campaign_name = sprintf( 'Test for %s', $this->get_campaign_name( $post ) );
+		$campaign      = $this->create_campaign( get_post( $post_id ), $campaign_name );
+		if ( is_wp_error( $campaign ) ) {
+			return $campaign;
 		}
-		$test_result = $this->api_request(
+		add_post_meta( $post_id, 'ac_test_campaign', $campaign['id'] );
+		$test_result = $this->api_v1_request(
 			'campaign_send',
 			'GET',
 			[
 				'query' => [
 					'type'       => 'html',
 					'action'     => 'test',
-					'campaignid' => $sync_result['campaign_id'],
+					'campaignid' => $campaign['id'],
 					'messageid'  => $sync_result['message_id'],
```

Comment:
`$sync_result` is not defined in this method, resulting in a warning:

```
PHP Notice:  Undefined variable: sync_result in /srv/users/user92978b3e/apps/user92978b3e/public/wp-content/plugins/newspack-newsletters/includes/service-providers/active_campaign/class-newspack-newsletters-active-campaign.php on line 300
```

Reply:
Nice catch! Fixed by syncing before send in 2c1322e5a1c1e8fb2e79327cb8250b61f7616046